python/src/trezorlib/_internal/emulator.py

```
@@ -437,6 +409,45 @@ def make_args(self) -> List[str]:
+    def ble_port(self) -> tuple[int, int]:
+        return (self.port + 4, self.port + 5)
+
+    # Tropic model can be managed externally, return configured port if set.
+    # Also TCP instead of UDP.
+    def tropic_port(self) -> int:
+        return self.tropic_model_port or (self.port + 6)
+
+    def properties(self) -> dict[str, str]:
+        args = [str(self.executable), "--emulator-properties"]
+        try:
+            completed = subprocess.run(
```

Ruff (0.15.20)

[error] `subprocess` call: check for execution of untrusted input

(S603)